### PR TITLE
add imu driver for db21m

### PR DESCRIPTION
**Pull request by jasonhu5**
## Description

Read IMU data on DB21Ms.

### How to test
1. `git clone -b daffy-db21m-imu-driver https://github.com/duckietown/dt-duckiebot-interface.git`
2. `cd [path/to/]dt-duckiebot-interface`
3. Build: `dts devel build -f --pull -H [HOSTNAME]`
4. Run:
    1. stop `duckiebot-interface` container from portainer
    2. `dts devel run -H [HOSTNAME] -M -s -f -- --privileged -it -v /tmp/argus_socket:/tmp/argus_socket`
5. Check:
    1. Open another terminal, `dts start_gui_tools [HOSTNAME]`
    2. `rostopic echo /[HOSTNAME]/imu_node/imu_data`
    3. Try giving the robot angular velocities and linear accelerations (manually or with `dts duckiebot keyboard_control [HOSTNAME]`) and verify with the `imu_data` messages (the previous step)

Typical msg from imu_data:
```
---                                                                                                                                                                                                        [20/17567]
header:                                                                                                                                                                                                              
  seq: 760                                                                                                                                                                                                           
  stamp:                                                                                                                                                                                                             
    secs: 1618840708                                                                                                                                                                                                 
    nsecs:  93475818                                                                                                                                                                                                 
  frame_id: ""                                                                                                                                                                                                  
orientation:                                                                                                                                                                                                         
  x: 0.0                                                                                                                                                                                                             
  y: 0.0                                                                                                                                                                                                             
  z: 0.0                                                                                                                                                                                                             
  w: 0.0                                                                                                                                                                                                             
orientation_covariance: [-1.0, -1.0, -1.0, -1.0, -1.0, -1.0, -1.0, -1.0, -1.0]                                                                                                                                       
angular_velocity:                                                                                                                                                                                                    
  x: 0.01638244032320685                                                                                                                                                                                             
  y: 0.009190149449603842                                                                                                                                                                                            
  z: -0.00532762286933556                                                                                                                                                                                            
angular_velocity_covariance: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]                                                                                                                                           
linear_acceleration:                                                                                                                                                                                                 
  x: 0.17002982905982905                                                                                                                                                                                             
  y: -0.19158290598290598                                                                                                                                                                                            
  z: -10.058102564102562
linear_acceleration_covariance: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
---
```

## Type of change

Please mark the relevant options with X:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change is constrained only to one package, any interfaces to nodes from other packages are maintained
- [ ] This change is constrained only to packages in this repository, any interfaces to nodes from other repositories are maintained
- [ ] This change required a documentation update
- [ ] This change is only a documentation update
- [ ] This change restructures the repository (at level higher than package level)
- [ ] This change requires a change in the Duckiebook (if yes, please describe and, if possible, make a pull request for an edit in the book and link it here)

## Tests

Please describe all tests ran and their outcome. It is highly recommended that you at least do the tests listed by default, even if you only changed the documentation.

- [ ] Made the Duckiebot move with keyboard control
- [ ] Visually checked the published camera stream
- [ ] Verified that the lane following demo works
- [ ] Verified that the indefinite navigation demo works

## Style compliance
Please make sure that all your changes conform to the Duckietown (code style)[link] and code (documentation style)[link] guidelines:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I checked that this pull request is compliant with the Duckietown code style
- [ ] I checked that this pull request is compliant with the Duckietown code documentation style

## Next steps
Please describe if you have any questions, comments, or doubts towards the Duckietown software guardians regarding implementation, code style, running tests, documentation, etc.

## FOR THE REVIEWERS
__IMPORTANT:__ DO NOT CHANGE THIS SECTION!

- [ ] Functionality (is the change necessary and is it implemented in the most appropriate way): @liampaull or @afdaniele
- [ ] Integration testing (verify the tests and run additional if necessary): @?
- [ ] Code style compliance: @gibernas
- [ ] Documentation compliance: @aleksandarpetrov

_To the reviewers:_ If you deem necessary, request code edits, more tests, documentation update, or additional reviewing. __Do not review your own code!__

_PR Template, last edit 23 Aug 2019_

### CODE REVIEW

**Comment by wagonhelm** [comment]
@afdaniele Is there anything I can do to help assist this PR make its way into Daffy?

File: packages/duckiebot_interface/launch/all_drivers.launch [hunk 1]
```diff
@@ -115,6 +115,13 @@
         </include>
     </group>
 
+    <!-- IMU sensor: -->
+    <group if="$(eval arg('robot_configuration') == 'DB21M')">
```
Comment (afdaniele) [inline packages/duckiebot_interface/launch/all_drivers.launch:119]: it should be 
```
if="$(eval arg('robot_configuration') in ['DB21M', 'DB21J'])"
```

File: packages/imu/src/imu_node.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            gyro_data = self._imu.gyro
+            self.logdebug('Gyro: {:.3f} {:.3f} {:.3f} dps'.format(*gyro_data))
+
+            try:
+                mag = self._imu.mag
+                self.logdebug('Magnet: {:.3f} {:.3f} {:.3f} mT'.format(*mag))
+                temp = self._imu.temp
+                self.logdebug('Temperature: {:.3f} C'.format(temp))
+            except:
+                self.logdebug("Didn't manage to read magnet/temp")
```
Comment (afdaniele) [inline packages/imu/src/imu_node.py:91]: A little too much debug, these are generating ROS messages on the `/rosout` topic.

**Review by afdaniele — CHANGES_REQUESTED** [review]

**Comment by frank-qcd-qk** [comment]
Temporarily closing and modifying target toward daffy staging. Syncing with daffy staging and addressing modifications.

**Comment by frank-qcd-qk** [comment]
Updated:
1. Use MPU 6050 adafruit driver
2. Added zero service call
3. Added stub TOF, IMU urdf. Pending CAD

**Pull request merged**